### ISSUE TITLE
[Feature Request] Audiobook chapters

### ISSUE THREAD

**Issue by mimmiepower** [flackbash/AudioAnchor#61]
The M4B format supports chapters and chapter images. iTunes does this for audiobooks. I would love if I were able to see which chapter I was on when listening to an audiobook and to easily hop between chapters. Possibly chapter markers and names popping up when scrobbling.

**Comment by flackbash** [comment]
I agree this would be a useful feature. I had been looking into this a while ago and it seemed after a quick search that it is surprisingly complicated to retrieve chapter information in Android, but I'll look into it again.

However, if I implement this, it will probably not be via markers on the seekbar as in the YouTube  chapter feature.  I think especially for long audio books with many chapters this would not be feasible without an additional seekbar-zooming feature (think e.g. a 20h audio book with 100+ chapters on a smartphone-sized screen).

Instead, the most feasible way is probably to implement this similar to the bookmarks feature. You would have a dialog with a list of chapters to choose from. Additionally, one could display the current chapter title somewhere on the screen.

**Comment by mimmiepower** [comment]
> 
> 
> I agree this would be a useful feature. I had been looking into this a while ago and it seemed after a quick search that it is surprisingly complicated to retrieve chapter information in Android, but I'll look into it again.
> 
> However, if I implement this, it will probably not be via markers on the seekbar as in the YouTube chapter feature. I think especially for long audio books with many chapters this would not be feasible without an additional seekbar-zooming feature (think e.g. a 20h audio book with 100+ chapters on a smartphone-sized screen).
> 
> Instead, the most feasible way is probably to implement this similar to the bookmarks feature. You would have a dialog with a list of chapters to choose from. Additionally, one could display the current chapter title somewhere on the screen.

Thank you for looking into it! I hope you'll find a solution.

**Comment by zouthie** [comment]
Hello. 

I just stumbled upon a .m4b file which has a companion .cue

**Comment by DearFox** [comment]
It's a shame this is still not supported. I downloaded the audiobook in mp3 and it's just a list of audio files without normal chapter titles. I could use m4b in place of this, but unfortunately this will not solve the problem :(

**Comment by oliwer** [comment]
FYI, [androidx.media3](https://github.com/androidx/media/tree/release/libraries/extractor/src/main/java/androidx/media3/extractor/mp4) supports MP4 metadata, including [audio book chapters](https://developer.android.com/reference/androidx/media3/common/MediaMetadata#MEDIA_TYPE_AUDIO_BOOK()).
